mapper/cumulocity/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -89,12 +89,15 @@ impl SmartRestUpdateSoftware {
         for module in &self.modules() {
             match module.action.clone().try_into()? {
                 CumulocitySoftwareUpdateActions::Install => {
+                    let url = module.get_url().map(|url| DownloadInfo::new(url.as_str()));
```

Comment:
One can have the `get_url()` method returning directly an `Option<DownloadInfo>`.